### PR TITLE
Remove hashes when converting http URL to ws URL

### PR DESCRIPTION
**Pull request by ericls**
## Description
Somehow I got a `#` in the URL of graphiQL. And subscription stops working because `ws://localhost:8000/graphql#` is not a valid URL for websocket.
So I wanted to remove hashes in URLs when converting for use in websockets. I'm not comfortable with using regex for that so I switched to using [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Make sure no hash part of URL is present when converting an URL for use with websocket


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Shen Li for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by DoctorJohn — APPROVED** [review]
Thanks for your contribution! I especially like the use of an URL object 👍 very elegant :)

File: strawberry/static/graphiql.html [hunk 1]
```diff
@@ -110,7 +110,11 @@
       const fetchURL = window.location.href;
 
       function httpUrlToWebSockeUrl(url) {
-        return url.replace(/(http)(s)?\:\/\//, "ws$2://");
+        const parsedURL = new URL(url);
+        const protocol = parsedURL.protocol === "http:" ? "ws:" : "wss:"
```
Comment (DoctorJohn) [inline strawberry/static/graphiql.html:114]: (if someone else reviewing this is wondering: yes, the protocols returned by `parsedURL.protocol` include a colon)

Comment (patrick91) [reply strawberry/static/graphiql.html:114]: that's odd! :D

**Comment by patrick91** [comment]
@ericls thanks for the PR!

I'll take the opportunity to also update GraphiQL, I'll send a commit here soon :)

**Comment by patrick91** [comment]
Codecov is failing, but it is probably just a glitch, will merge now! Thanks again!

**Comment by patrick91** [comment]
The bot is broken, so I'll post the message manually :)

---

Thanks for contributing to Strawberry! 🎉 You've been invited to join the Strawberry GraphQL organisation 😊
  
You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67
  
And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**